qui/tray/disk_space.py

```diff
@@ -160,31 +195,29 @@ def __populate_pools(self):
         except exc.QubesDaemonAccessError:
             pools = []
         for pool in pools:
-            self.pools.append(pool)
-            if not getattr(pool, 'size', None) or \
-                    'included_in' in getattr(pool, 'config', {}):
+            wrapped_pool = PoolWrapper(pool)
+            self.pools.append(wrapped_pool)
+
+            if wrapped_pool.has_error:
                 continue
-            self.total_size += getattr(pool, 'size', 0)
-            self.used_size += getattr(pool, 'usage', 0)
-            try:
-                if pool.usage/pool.size >= URGENT_WARN_LEVEL:
-                    self.warning_message.append(
-                        _("\n{:.1%} space left in pool {}").format(
-                            1-pool.usage/pool.size, pool.name))
-            except (ValueError, exc.QubesDaemonAccessError):
-                pass
-            try:
-                if pool.usage_details.get('metadata_size', None):
-                    metadata_usage = pool.usage_details['metadata_usage'] / \
-                                     pool.usage_details['metadata_size']
-                    if metadata_usage >= URGENT_WARN_LEVEL:
-                        # pylint: disable=consider-using-f-string
-                        self.warning_message.append(_(
-                            "\nMetadata space for pool {} is running out. "
-                            "Current usage: {.1%}").format(
-                                pool.name, metadata_usage))
-            except (exc.QubesPropertyAccessError, AttributeError):
-                pass
+
+            if not wrapped_pool.size or not wrapped_pool.usage or \
+                    'included_in' in wrapped_pool.config:
+                continue
+            self.total_size += wrapped_pool.size
+            self.used_size += wrapped_pool.usage
+
+            if wrapped_pool.usage_perc >= URGENT_WARN_LEVEL:
+                self.warning_message.append(
+                    _("\n{:.1%} space left in pool {}").format(
+                        1-wrapped_pool.usage_perc, wrapped_pool.name))
+
+            if wrapped_pool.metadata_usage >= URGENT_WARN_LEVEL:
```

Comment:
metadata_perc?

Currently it fails this way for pools without metadata usage:
```
May 23 10:05:41 dom0 widget-wrapper[3024]: Traceback (most recent call last):
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/bin/qui-disk-space", line 33, in <module>
May 23 10:05:41 dom0 widget-wrapper[3024]:     sys.exit(load_entry_point('qui==0.1', 'gui_scripts', 'qui-disk-space')())
May 23 10:05:41 dom0 widget-wrapper[3024]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 506, in main
May 23 10:05:41 dom0 widget-wrapper[3024]:     app = DiskSpace()
May 23 10:05:41 dom0 widget-wrapper[3024]:           ^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 355, in __init__
May 23 10:05:41 dom0 widget-wrapper[3024]:     self.refresh_icon()
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 362, in refresh_icon
May 23 10:05:41 dom0 widget-wrapper[3024]:     pool_data = PoolUsageData(self.qubes_app)
May 23 10:05:41 dom0 widget-wrapper[3024]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 190, in __init__
May 23 10:05:41 dom0 widget-wrapper[3024]:     self.__populate_pools()
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 215, in __populate_pools
May 23 10:05:41 dom0 widget-wrapper[3024]:     if wrapped_pool.metadata_usage >= URGENT_WARN_LEVEL:
May 23 10:05:41 dom0 widget-wrapper[3024]:        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]: TypeError: '>=' not supported between instances of 'NoneType' and 'float'
```

Reply:
you're right, pushed a fix